spec/src/test/java/impl/QuestionResourceTest.java

```diff
@@ -100,38 +102,113 @@ public void shouldListLatest10QuestionsAsDefault() {
+
+        // given a question and a answer
+        Question question = createQuestionAndAnswer(mockAuth);
+
+        // when searching with a query that matches the body of the answer
+        List<Question> questions = questionResource.getQuestionsBySearchQuery("Answer body").toBlocking().single();
+
+        // then the question should be returned that matches the answer (case insensitive )
+        assertEquals(1, questions.size());
+        assertThat(questions.get(0).getTitle()).isEqualTo(question.getTitle());
+    }
+
+    private Question createQuestionAndAnswer(Auth mockAuth) {
         // given a question
         Question question = TestSetup.getQuestion("Question title", "Question");
         questionResource.createQuestion(mockAuth, question).toBlocking().singleOrDefault(null);
         // and an answer
         Answer answer = TestSetup.getAnswer("Answer body");
         answerResource.answerQuestion(mockAuth, answer, question.getId()).toBlocking().singleOrDefault(null);
+        return question;
+    }
 
-        // when searching with a query that matches the body of the answer
-        List<Question> questions = questionResource.getQuestionsBySearchQuery("body").toBlocking().single();
-        // then the question should be returned
-        assertEquals(1, questions.size());
-        assertThat(questions.get(0).getTitle()).isEqualTo(question.getTitle());
+    @Test
+    public void shouldReturnEmptyResultIfSearchQueryIsEmpty() {
+        Auth mockAuth = createUserAndAuth();
+
+        // given a question and a answer
+        createQuestionAndAnswer(mockAuth);
+
+        // when searching with non matching query
+        List<Question> questions = questionResource.getQuestionsBySearchQuery("no results for this query").toBlocking().single();
+
+        // then no questions should be returned
+        assertThat(questions).isEmpty();
     }
 
     @Test
-    public void shouldNotBeAbleToSearchEmptyOrNull() {
-        Observable<List<Question>> questions = questionResource.getQuestionsBySearchQuery("");
+    public void shouldReturnEmptyIfSearchQueryIsEmpty() {
+        Auth mockAuth = createUserAndAuth();
+
+        // given a question and a answer
+        createQuestionAndAnswer(mockAuth);
+
+        // when searching with non matching query
+        List<Question> questions = questionResource.getQuestionsBySearchQuery("").toBlocking().single();
+
+        // then no questions should be returned
+        assertThat(questions).isEmpty();
+    }
+
+    @Test
+    public void shouldReturnEmptyIfSearchQueryIsNull() {
+        Auth mockAuth = createUserAndAuth();
+
+        // given a question and a answer
+        createQuestionAndAnswer(mockAuth);
+
+        // when searching with no query
+        List<Question> questions = questionResource.getQuestionsBySearchQuery(null).toBlocking().single();
+
+        // then no questions should be returned
+        assertThat(questions).isEmpty();
```

Comment:
this is, as stated in the description, really something that might be wrong. If you look at it in a strict way. For me , this was more a pragmatic way.  The true brutal way, would be to throw an exception here. But one could argue that query params are optional from a http perspective and not using any should return everything. That would be correct i guess, but not for the client. Im thinking that a api like this should be formed by the ui, as we do with questions and answers. We are including the users name etc. So its really about opinion i guess.

Reply:
There are possibly two variants of 400 Bad Request that can be used here. One indicating that the query hasn't been supplied (missing parameter) and one that a query has been supplied but it's syntax is invalid (such as the empty string). Im guessing that the first error can be avoided by having a default query run when the parameter is missing. The second error should be reported to the client - if a ui is providing the request a code error could lead to the query parameter not being supplied. Returning an empty list here could be misleading.